### PR TITLE
Add support for managing IAM Trust Relationships

### PR DESCRIPTION
**Pull request by diranged**
CC @Siminm

### CODE REVIEW

**Comment by nextdoor-engineer** [comment]
Looks good. Thanks

File: kingpin/actors/aws/iam/test/test_entities.py [hunk 1]
```diff
@@ -738,19 +739,95 @@ def setUp(self):
...
+    def test_ensure_assume_role_doc_no_entity(self):
+        fake_entity = None
+        self.actor._get_entity = mock.MagicMock()
+        self.actor._get_entity.side_effect = [tornado_value(fake_entity)]
+
+        yield self.actor._ensure_assume_role_doc('test')
+
+    @testing.gen_test
+    def test_ensure_assume_role_doc_matches(self):
+        lambda_string = (
```
Comment (siminm) [inline kingpin/actors/aws/iam/test/test_entities.py]: Not readable :(

A bit more work, but can easily do this here:

```python
request = {
  "Version": "2012-10-17",
  "Statement": [{"Action": ["s3:Create*", "s3:Get*", "s3:Put*", "s3:List*"],
                 "Resource": ["arn:aws:s3:::kingpin*/*", "arn:aws:s3:::kingpin*"],
                 "Effect": "Allow"}]}
lambda_string = urllib.pathname2url( json.dumps( request ) )
```

File: kingpin/actors/aws/iam/test/test_entities.py [hunk 2]
```diff
@@ -723,7 +723,8 @@ def setUp(self):
             'Unit Test',
```
Comment (siminm) [inline kingpin/actors/aws/iam/test/test_entities.py]: Convert to same style as line 751

File: kingpin/actors/aws/iam/test/test_entities.py (hunk 1 above)
Comment (diranged) [inline kingpin/actors/aws/iam/test/test_entities.py]: Bah .. :P Ok done.

**Comment by siminm** [comment]
👍 when tests pass

**Pull request merged**